avocado/utils/network/interfaces.py

```diff
@@ -388,52 +400,92 @@ def save(self, ipaddr, netmask):
+            if os.path.exists(f"{path}/{filename}"):
+                pass
+            else:
+                cmd1 = f"nmcli connection add con-name {self.name} ifname {self.name} type ethernet ipv4.address {ipaddr}/{prefix}"
+                run_command(cmd1, self.host)
+            cmd2 = f"nmcli c mod id {self.name} ipv4.method manual ipv4.address {ipaddr}/{prefix}"
+            cmd3 = f"nmcli connection up {self.name}"
+            run_command(cmd2, self.host)
+            run_command(cmd3, self.host)
+            self._move_file_to_backup(f"{path}/{filename}")
+            if os.path.exists(f"{path}/{filename}.backup"):
+                destination = f"{path}/{filename}"
+                shutil.copy(f"{path}/{filename}.backup", destination)
+        else:
+            ifcfg_dict = {'TYPE': self.if_type,
+                          'BOOTPROTO': 'static',
+                          'NAME': self.name,
+                          'DEVICE': self.name,
+                          'ONBOOT': 'yes',
+                          'IPADDR': ipaddr,
+                          'NETMASK': netmask,
+                          'IPV6INIT': 'yes',
+                          'IPV6_AUTOCONF': 'yes',
+                          'IPV6_DEFROUTE': 'yes'}
+            if current_distro.name == 'SuSE':
+                ifcfg_dict = {'IPADDR': ipaddr,
+                              'BOOTPROTO': 'static',
+                              'STARTMODE': 'auto'}
+            self._write_to_file(f"{path}/{filename}", ifcfg_dict)
 
         if self.if_type == "Bond":
-            ifcfg_dict["BONDING_MASTER"] = "yes"
             bond_dict = self._get_bondinterface_details()
-            ifcfg_slave_dict = {"SLAVE": "yes", "ONBOOT": "yes", "MASTER": self.name}
-            if current_distro.name == "SuSE":
-                ifcfg_dict["BONDING_MODULE_OPTS"] = "mode=" + bond_dict["mode"][0]
-                for index, slave in enumerate(bond_dict["slaves"]):
-                    bonding_slave = f"BONDING_SLAVE{index}"
-                    ifcfg_dict[bonding_slave] = slave
-                    ifcfg_slave_dict.update({"NAME": slave, "DEVICE": slave})
-                    self._write_to_file(f"{path}/ifcfg-{slave}", ifcfg_slave_dict)
-            elif current_distro.name in ["rhel", "fedora"]:
-                ifcfg_dict["BONDING_OPTS"] = "mode=" + bond_dict["mode"][0]
-                for index, slave in enumerate(bond_dict["slaves"]):
-                    ifcfg_slave_dict.update(
-                        {"NAME": slave, "DEVICE": slave, "TYPE": "Ethernet"}
-                    )
-                    self._write_to_file(f"{path}/ifcfg-{slave}", ifcfg_slave_dict)
+            if current_distro.name == 'rhel' and current_distro.version == '9':
+                if os.path.exists(f"{path}/{filename}"):
+                    pass
+                else:
+                    cmd1 = f"nmcli connection add con-name {self.name} ifname {self.name} type ethernet ipv4.address {ipaddr}/{prefix}"
```

Comment:
is nmcli shipped with default distro if not we need to have check for this

Reply:
@PraveenPenguin  The check is already present for distro versions , then based on the versions nmconnections are done using nmcli commands, Else to support old distro we already have dictionary/old method

Reply:
so nmcli is shipped default with distro ?